shared/constants/network.ts

```diff
@@ -0,0 +1,579 @@
+});
+export const RINKEBY_RPC_URL = getRpcUrl({
+  network: NETWORK_TYPES.RINKEBY,
+});
+export const KOVAN_RPC_URL = getRpcUrl({ network: NETWORK_TYPES.KOVAN });
+export const MAINNET_RPC_URL = getRpcUrl({
+  network: NETWORK_TYPES.MAINNET,
+});
+export const GOERLI_RPC_URL = getRpcUrl({ network: NETWORK_TYPES.GOERLI });
+export const LOCALHOST_RPC_URL = 'http://localhost:8545';
+
+/**
+ * An object containing the token symbols for various tokens that are either
+ * native currencies or those that have been special cased by the extension
+ * for supporting our feature set.
+ */
+export const CURRENCY_SYMBOLS = {
+  ARBITRUM: 'AETH',
+  AVALANCHE: 'AVAX',
+  BNB: 'BNB',
+  BUSD: 'BUSD',
+  CELO: 'CELO',
+  DAI: 'DAI',
+  ETH: 'ETH',
+  FANTOM: 'FTM',
+  HARMONY: 'ONE',
+  PALM: 'PALM',
+  MATIC: 'MATIC',
+  TEST_ETH: 'TESTETH',
+  USDC: 'USDC',
+  USDT: 'USDT',
+  WETH: 'WETH',
+} as const;
+
+/**
+ * A type that is a union type of all possible hardcoded currency symbols.
+ * This type is non-exhaustive, and cannot be used for areas where the user
+ * or dapp may supply their own symbol.
+ */
+type CurrencySymbol = typeof CURRENCY_SYMBOLS[keyof typeof CURRENCY_SYMBOLS];
+/**
+ * For certain specific situations we need the above type, but with all symbols
+ * in lowercase format.
+ */
+type LowercaseCurrencySymbol = Lowercase<CurrencySymbol>;
+
+export const ETH_TOKEN_IMAGE_URL = './images/eth_logo.svg';
+export const TEST_ETH_TOKEN_IMAGE_URL = './images/black-eth-logo.svg';
+export const BNB_TOKEN_IMAGE_URL = './images/bnb.png';
+export const MATIC_TOKEN_IMAGE_URL = './images/matic-token.png';
+export const AVAX_TOKEN_IMAGE_URL = './images/avax-token.png';
+export const AETH_TOKEN_IMAGE_URL = './images/arbitrum.svg';
+export const FTM_TOKEN_IMAGE_URL = './images/fantom-opera.svg';
+export const HARMONY_ONE_TOKEN_IMAGE_URL = './images/harmony-one.svg';
+export const OPTIMISM_TOKEN_IMAGE_URL = './images/optimism.svg';
+export const PALM_TOKEN_IMAGE_URL = './images/palm.svg';
+
+export const INFURA_PROVIDER_TYPES = [
+  NETWORK_TYPES.ROPSTEN,
+  NETWORK_TYPES.RINKEBY,
+  NETWORK_TYPES.KOVAN,
+  NETWORK_TYPES.MAINNET,
+  NETWORK_TYPES.GOERLI,
+];
+
+export const TEST_CHAINS = [
+  CHAIN_IDS.ROPSTEN,
+  CHAIN_IDS.RINKEBY,
+  CHAIN_IDS.GOERLI,
+  CHAIN_IDS.KOVAN,
+  CHAIN_IDS.LOCALHOST,
+];
+
+export const TEST_NETWORK_TICKER_MAP = {
+  [NETWORK_TYPES.ROPSTEN]: `${
+    capitalize(NETWORK_TYPES.ROPSTEN) as Capitalize<
+      typeof NETWORK_TYPES.ROPSTEN
+    >
+  }${CURRENCY_SYMBOLS.ETH}`,
+  [NETWORK_TYPES.RINKEBY]: `${
+    capitalize(NETWORK_TYPES.RINKEBY) as Capitalize<
+      typeof NETWORK_TYPES.RINKEBY
+    >
+  }${CURRENCY_SYMBOLS.ETH}`,
+  [NETWORK_TYPES.KOVAN]: `${
+    capitalize(NETWORK_TYPES.KOVAN) as Capitalize<typeof NETWORK_TYPES.KOVAN>
+  }${CURRENCY_SYMBOLS.ETH}`,
+  [NETWORK_TYPES.GOERLI]: `${
+    capitalize(NETWORK_TYPES.GOERLI) as Capitalize<typeof NETWORK_TYPES.GOERLI>
+  }${CURRENCY_SYMBOLS.ETH}`,
+} as const;
```

Comment:
Could you also write this as the following?

``` suggestion
export const TEST_NETWORK_TICKER_MAP: {
  [K in NetworkTypes]: Capitalize<K>
}: Object.keys(NETWORK_TYPES).reduce((object, key) => {
  return { ...object, [key]: capitalize(key) };
}, {});
```

Reply:
Upon further investigation, the proposed would not result in what we want.

We would need to create a new variable that contained only the test nets we want to reduce on, otherwise we'd need to do logic checks in the reduce method to extract the non-test-nets, we'd also need to make the Capitalize<K> part be `${Capitalize<K>}ETH` which Is fine but overall the complexity doesn't save much over just being explicit. For brevity I changed to this instead:

```
const typedCapitalize = <K extends string>(k: K): Capitalize<K> =>
  capitalize(k) as Capitalize<typeof k>;

export const TEST_NETWORK_TICKER_MAP: {
  [K in Exclude<
    NetworkTypes,
    'localhost' | 'mainnet' | 'rpc'
  >]: `${Capitalize<K>}${typeof CURRENCY_SYMBOLS.ETH}`;
} = {
  [NETWORK_TYPES.ROPSTEN]: `${typedCapitalize(NETWORK_TYPES.ROPSTEN)}${
    CURRENCY_SYMBOLS.ETH
  }`,
  [NETWORK_TYPES.RINKEBY]: `${typedCapitalize(NETWORK_TYPES.RINKEBY)}${
    CURRENCY_SYMBOLS.ETH
  }`,
  [NETWORK_TYPES.KOVAN]: `${typedCapitalize(NETWORK_TYPES.KOVAN)}${
    CURRENCY_SYMBOLS.ETH
  }`,
  [NETWORK_TYPES.GOERLI]: `${typedCapitalize(NETWORK_TYPES.GOERLI)}${
    CURRENCY_SYMBOLS.ETH
  }`,
} as const;
```